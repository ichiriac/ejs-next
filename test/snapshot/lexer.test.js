const lexer = require('../../lib/parser/lexer');

describe('Lexer', () => {
  it('Simple case', () => {
    var lex = new lexer();
    lex.input(`
    <foo>
    <%# comment %>
    <%_ if (foo == "%>\\"%>") { _%>
      <%= bar %>
    <% } %>
    </foo>
    `);
    var tokens = [];
    while(true) {
      var tok = lex.next();
      tokens.push(tok);
      if (tok[0] === lexer.tokens.T_EOF) break;
    }
    expect(tokens).toMatchSnapshot();
  });
});