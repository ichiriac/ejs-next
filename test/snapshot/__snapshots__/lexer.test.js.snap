// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lexer Simple case 1`] = `
Array [
  Array [
    1,
    "
    <foo><%% 
    ",
    0,
    19,
  ],
  Array [
    2,
    "<%",
    19,
    21,
  ],
  Array [
    4,
    "#",
    21,
    22,
  ],
  Array [
    8,
    " comment ",
    22,
    31,
  ],
  Array [
    3,
    "%>",
    31,
    33,
  ],
  Array [
    1,
    "
    ",
    33,
    38,
  ],
  Array [
    2,
    "<%",
    38,
    40,
  ],
  Array [
    7,
    "_",
    40,
    41,
  ],
  Array [
    8,
    " if (foo == \\"%>\\\\\\"%>\\") { ",
    41,
    65,
  ],
  Array [
    7,
    "_",
    65,
    66,
  ],
  Array [
    3,
    "%>",
    66,
    68,
  ],
  Array [
    1,
    "
      ",
    68,
    75,
  ],
  Array [
    2,
    "<%",
    75,
    77,
  ],
  Array [
    5,
    "=",
    77,
    78,
  ],
  Array [
    8,
    " bar ",
    78,
    83,
  ],
  Array [
    3,
    "%>",
    83,
    85,
  ],
  Array [
    1,
    "
    ",
    85,
    90,
  ],
  Array [
    2,
    "<%",
    90,
    92,
  ],
  Array [
    8,
    " } ",
    92,
    95,
  ],
  Array [
    3,
    "%>",
    95,
    97,
  ],
  Array [
    1,
    "
    </foo>
    ",
    97,
    113,
  ],
  Array [
    0,
    "",
    113,
    113,
  ],
]
`;
